integration/entity.go

```diff
@@ -43,7 +48,13 @@ func newLocalEntity(storer persist.Storer, addHostnameToMetadata bool) *Entity {
 }
 
 // newEntity creates a new remote-entity.
-func newEntity(name, namespace string, storer persist.Storer, addHostnameToMetadata bool) (*Entity, error) {
+func newEntity(
+	name,
+	namespace string,
+	storer persist.Storer,
+	addHostnameToMetadata bool,
+	opts ...entityOption,
```

Comment:
I won't use func options here because:
- this won't ease customer usage as it's private
- this won't ease sdk-dev usage as there are no defined funcs, so you need to follow the whole flow to understand what's being done
- it's not consistent with rest of the options, either all func-opts, either none